dashboard/app/helpers/levels_helper.rb

```diff
@@ -237,16 +237,21 @@ def blockly_options(local_assigns={})
       open_function_definition
       callout_json
       disable_sharing
+      hide_source
+      share
+      no_padding
+      show_finish
+      embed
     ).map{ |x| x.include?(':') ? x.split(':') : [x,x.camelize(:lower)]}]
     .each do |dashboard, blockly|
       # Select first valid value from 1. local_assigns, 2. property of @level object, 3. named instance variable, 4. properties json
       # Don't override existing valid (non-nil/empty) values
-      property = local_assigns[dashboard].presence ||
+      property = local_assigns[dashboard.to_sym].presence ||
         @level[dashboard].presence ||
         instance_variable_get("@#{dashboard}").presence ||
-        level[dashboard.to_s].presence
-      value = blockly_value(level[blockly.to_s] || property)
-      level[blockly.to_s] = value unless value.nil? # make sure we convert false
+        level[dashboard].presence
+      value = blockly_value(level[blockly] || property)
```

Comment:
Do you know what bugs come from `blockly_value`?  I think this is needed because properties are stored as strings but `1.0` and `false` need to converted to unquoted values.  When switching XML->JSON it probably makes sense to store properties as the correct type.  I'll look at making the switch during that refactor.